### PR TITLE
fix: allow overriding of StackOverflow configuration

### PR DESCRIPTION
**Pull request by ScottGuymer**
## Hey, I just made a Pull Request!

I am currently working on integrating our backstage instance with a private stackoverflow instance.

To make the frontend work I have proxied the private SO API via the proxy functionality. 

But for the backend this means that it will try to go out and hit the public DNS and will require authentication. 

To combat this I want to be able to configure the proxy URL for the frontend and the public URL for the backend. Therefor I want to be able to override the config when instantiating the search indexer. Like this 

```
  indexBuilder.addCollator({
    schedule,
    factory: StackOverflowQuestionsCollatorFactory.fromConfig(env.config, {
      logger: env.logger,
      baseUrl: env.config.get('stackoverflow.backendBaseUrl'),
      requestParams: {
        tagged: ['my-tag'],
        pagesize: 100,
        order: 'desc',
        sort: 'activity',
      },
    }),
  });
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by ScottGuymer** [comment]
@iamEAP looks like you recently approved something in this area.

**Review by aitherios — APPROVED** [review]
Thank you for this PR! Specially creating the tests for `StackOverflowQuestionsCollatorFactory`.

File: plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.test.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      defaultOptions,
+    );
+    expect(factory.type).toBe('stack-overflow');
+  });
+
+  describe('getCollator', () => {
+    const worker = setupServer();
+    setupRequestMockHandlers(worker);
+
+    afterEach(async () => {
```
[the remark this answers is not in the document]
Comment (ScottGuymer) [reply plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.test.ts:98]: Ah interesting. 

I was getting warnings about open handles from jest and I tried this to try and prevent that (with no luck, still got the warning)

I will remove it.

**Comment by ScottGuymer** [comment]
Not sure why the test run failed. Doesnt seem at all related to the changes i have made.

**Comment by ScottGuymer** [comment]
hi @aitherios this seems to be failing on a single CI run but not for anything i have changed.

Any chance you can re-start the CI?

**Comment by freben** [comment]
I gave it a nudge!

**Review by aitherios — APPROVED** [review]

**Comment by ScottGuymer** [comment]
Great thanks!

Is this good to merge then?

**Comment by ScottGuymer** [comment]
@aitherios @freben is this ok to merge?

**Comment by freben** [comment]
Checking!

**Review by freben — APPROVED** [review]
Thanks!

**Pull request merged**